### PR TITLE
Popover fixes

### PR DESCRIPTION
**Pull request by pleek91**
## PR Checklist

(_all items must be checked off for a PR to be merged_)

This PR:

- [x] has a changelog entry with a short description of what's changed in `CHANGELOG.md`
- [x] adds new tests or updates existing tests (or hasn't, for reasons explained below)
- [x] strictly follows the design spec (if one exists)
- [x] adheres to [a11y guidelines](https://www.a11yproject.com/checklist/)/[WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/) guidelines where applicable; this can be tested using Lighthouse (native in Chrome dev tools or as a plugin for other browsers) and with accessibility reports in Firefox

This PR has been tested in the following browsers (the latest version of each is fine):

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari

## PR description

- Updates popovers to use the position utilities used by tooltips and selects
- Fixes bug caused by using `this.$watch` in a class component that gets transpiled to a composition
- Add leftTop and rightTop positioning methods to the position utilities.

### CODE REVIEW

File: src/components/Popover/Popover.spec.js [hunk 1]
```diff
@@ -3,27 +3,38 @@ import { nextTick } from 'vue'
...
       }
     })
 
     wrapper.vm.openPopover()
+
+    // magical number of nextTicks. 
+    // todo: refactor openPopover to be async
+    await nextTick()
+    await nextTick()
     await nextTick()
```
Comment (stackoverfloweth) [inline src/components/Popover/Popover.spec.js:32]: 😞

File: src/utilities/strings.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export function snakeCase(string: string): string {
+    return string
+        .replace(/([a-z])([A-Z])/g, "$1-$2")
```
Comment (stackoverfloweth) [inline src/utilities/strings.ts:3]: it appears here like you’re putting in a ‘-’ where you maybe meant to put in a ‘_’

Comment (pleek91) [reply src/utilities/strings.ts:3]: that's supposed to be kebab case. Fixed 👍

File: src/components/Popover/Popover.spec.js (hunk 1 above)
Comment (pleek91) [reply src/components/Popover/Popover.spec.js:32]: I'm optimistic this can be fixed. But I don't have time ATM

**Pull request merged**